### PR TITLE
clang-tidy and build misc

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
Summary
 - semaphore-ci now responsible for compiling all builds and deploying to s3
 - travis-ci is for tests and static analysis
 - circleci OSX build and test

misc 
 - builds now running clang static analysis and cppcheck
 - travis-ci uploading tests code coverage to s3
 - travis-ci should be much faster
 - clang-tidy now only lists checks that are skipped
 - coverity setup improved

clang static analysis output: https://s3.amazonaws.com/px4-travis/Firmware/build_misc/scan-build/scan-build/index.html
cppcheck static analysis: https://s3.amazonaws.com/px4-travis/Firmware/build_misc/cppcheck/cppcheck/index.html
test code coverage: https://s3.amazonaws.com/px4-travis/Firmware/build_misc/coverage/coverage-html/index.html
coverity: https://scan.coverity.com/projects/px4-firmware

**Comment by dagar** [comment]
@bkueng can you skim this to review the code changes?

File: cmake/common/px4_base.cmake [hunk 1]
```diff
@@ -846,6 +846,15 @@ function(px4_add_common_flags)
 		)
 	endif()
 
+	if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+		if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.9)
+			# force color for gcc > 4.9
```
Comment (bkueng) [inline cmake/common/px4_base.cmake]: Nice!

File: src/lib/mathlib/math/Expo.hpp [hunk 2]
```diff
@@ -48,32 +48,33 @@ namespace math
 {
 
 // Type-safe signum function
-template<typename T> int sign(T val) {
+template<typename T> int sign(T val)
```
Comment (bkueng) [inline src/lib/mathlib/math/Expo.hpp]: While at it can we do:
```
template<typename T>
int sign(T val)
```

**Comment by LorenzMeier** [comment]
Too much churn for v1.6 but great work! Will merge towards end of this week.

File: src/platforms/posix/drivers/gyrosim/gyrosim.cpp [hunk 3]
```diff
@@ -312,7 +312,6 @@ GYROSIM::GYROSIM(const char *path_accel, const char *path_gyro, enum Rotation ro
 	VirtDevObj("GYROSIM", path_accel, ACCEL_BASE_DEVICE_PATH, 1e6 / 400),
 	_gyro(new GYROSIM_gyro(this, path_gyro)),
 	_product(GYROSIMES_REV_C4),
-	_call{},
```
Comment (bkueng) [inline src/platforms/posix/drivers/gyrosim/gyrosim.cpp]: Where does this change come from? I don't see `_call` being referenced at all, so should be safe to remove.

Comment (dagar) [inline src/platforms/posix/drivers/gyrosim/gyrosim.cpp]: This was done automatically by clang-tidy (removing redundant init). I'll remove the rest of the unused work handle.

**Pull request merged**